mathesar/tests/api/test_record_api.py

```diff
@@ -1277,6 +1277,42 @@ def test_number_input_api_validation(empty_nasa_table, client):
         assert response.status_code == status_code
 
 
+def test_invalid_email_post_api_validation(empty_nasa_table, client):
+    table = empty_nasa_table
+    column_name = 'Email'
+    table.add_column({"name": column_name, "type": 'mathesar_types.email'})
+    email_col_id = table.get_column_name_id_bidirectional_map()[column_name]
+    data = {
+        email_col_id: 'foobar'
+    }
+    response = client.post(f'/api/db/v0/tables/{table.id}/records/', data=data)
+    response_data = response.json()
+    assert response.status_code == 400
+    assert response_data[0]['code'] == ErrorCodes.CheckViolation.value
+    assert response_data[0]['message'] == 'The requested insert violates a check constraint'
+    assert response_data[0]['detail'] == 'value for domain mathesar_types.email violates check constraint \"email_check\"\n'
```

Comment:
The error can be rephrased as 
```suggestion
    assert response_data[0]['detail'] == 'Value for the column with type  mathesar_types.email violates check constraint \"email_check\"\n'
```

Reply:
Actually, this is the original error message generated by postgres that is being passed on to the API, should it be altered?